pkg/protocol/chains.go

```diff
@@ -0,0 +1,180 @@
+package protocol
+
+import (
+	"cmp"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+
+	"github.com/iotaledger/hive.go/ds/reactive"
+	"github.com/iotaledger/hive.go/lo"
+	"github.com/iotaledger/hive.go/log"
+	"github.com/iotaledger/iota-core/pkg/model"
+	"github.com/iotaledger/iota-core/pkg/protocol/engine"
+	iotago "github.com/iotaledger/iota.go/v4"
+)
+
+// Chains is a subcomponent of the protocol that exposes the chains that are managed by the protocol and that implements
+// the chain switching logic.
+type Chains struct {
+	// Set contains all non-evicted chains that are managed by the protocol.
+	reactive.Set[*Chain]
+
+	// Main contains the main chain.
+	Main reactive.Variable[*Chain]
+
+	// HeaviestClaimedCandidate contains the candidate chain with the heaviest claimed weight.
```

Comment:
These comments are somewhat useless: you are describing it with the same words. It would be better to describe it differently: 
```suggestion
	// HeaviestClaimedCandidate contains the candidate chain with the heaviest claimed weight according to its latest commitment. The weight has neither been checked via attestations nor verified by downloading all data.
```

Reply:
Most of the comments are AI generated and completely up for discussion. I mostly focused on removing all the unnecessary quirks in the code that required additional explanations but to be honest, we should most probably even discuss the naming of the properties.

Maybe sth like "HeaviestUnverifiedCandidate" would already much better to convey the idea of what kind of "knowledge" this property is supposed to capture,